tables/vpc_flow_log/vpc_flow_log_table.go

```diff
@@ -1,138 +1,365 @@
+		t, err := helpers.ParseTime(startTime)
+		if err != nil {
+			invalidFields = append(invalidFields, "start_time")
+		} else {
+			row.OutputColumns[constants.TpTimestamp] = t
+		}
+	} else if endTime, ok := row.GetSourceValue("end_time"); ok && (endTime != VpcFlowLogTableSkippedData || endTime != VpcFlowLogTableNoData || endTime != VpcFlowLogTableNilValue) {
+		t, err := helpers.ParseTime(endTime)
+		if err != nil {
+			invalidFields = append(invalidFields, "end_time")
+		} else if row.OutputColumns[constants.TpTimestamp] == nil {
+			row.OutputColumns[constants.TpTimestamp] = t
+		}
+	}
 
-	// Record standardization
-	row.TpID = xid.New().String()
-	row.TpIngestTimestamp = time.Now()
+	if len(invalidFields) > 0 {
+		return nil, error_types.NewRowErrorWithFields([]string{}, invalidFields)
+	}
 
-	// ips
-	if row.SrcAddr != nil {
-		row.TpSourceIP = row.SrcAddr
-		row.TpIps = append(row.TpIps, *row.SrcAddr)
+	// tp_ips
+	var ips []string
+	if srcAddr, ok := row.GetSourceValue("src_addr"); ok && (srcAddr != VpcFlowLogTableSkippedData || srcAddr != VpcFlowLogTableNoData || srcAddr != VpcFlowLogTableNilValue) {
+		ips = append(ips, srcAddr)
+		row.OutputColumns[constants.TpSourceIP] = srcAddr
+	}
+	if pktSrcAddr, ok := row.GetSourceValue("pkt_src_addr"); ok && (pktSrcAddr != VpcFlowLogTableSkippedData || pktSrcAddr != VpcFlowLogTableNoData || pktSrcAddr != VpcFlowLogTableNilValue) {
+		ips = append(ips, pktSrcAddr)
 	}
-	if row.PktSrcAddr != nil {
-		row.TpIps = append(row.TpIps, *row.PktSrcAddr)
+	if dstAddr, ok := row.GetSourceValue("dst_addr"); ok && (dstAddr != VpcFlowLogTableSkippedData || dstAddr != VpcFlowLogTableNoData || dstAddr != VpcFlowLogTableNilValue) {
+		ips = append(ips, dstAddr)
+		row.OutputColumns[constants.TpDestinationIP] = dstAddr
 	}
-	if row.DstAddr != nil {
-		row.TpDestinationIP = row.DstAddr
-		row.TpIps = append(row.TpIps, *row.DstAddr)
+	if pktDstAddr, ok := row.GetSourceValue("pkt_dst_addr"); ok && (pktDstAddr != VpcFlowLogTableSkippedData || pktDstAddr != VpcFlowLogTableNoData || pktDstAddr != VpcFlowLogTableNilValue) {
+		ips = append(ips, pktDstAddr)
 	}
-	if row.PktDstAddr != nil {
-		row.TpIps = append(row.TpIps, *row.PktDstAddr)
+	if len(ips) > 0 {
+		row.OutputColumns[constants.TpIps] = ips
 	}
 
-	// TpIndex
-	if row.InterfaceID != nil {
-		row.TpIndex = *row.InterfaceID
-	} else if row.SubnetID != nil {
-		row.TpIndex = *row.SubnetID
-	} else if row.VPCID != nil {
-		row.TpIndex = *row.VPCID
-	} else {
-		row.TpIndex = "default"
+	// tp_index
+	for _, key := range []string{"interface_id", "subnet_id", "vpc_id"} {
+		if val, ok := row.GetSourceValue(key); ok && (val != VpcFlowLogTableSkippedData || val != VpcFlowLogTableNoData || val != VpcFlowLogTableNilValue) {
+			row.OutputColumns[constants.TpIndex] = val
+			break
+		}
+	}
+	if row.OutputColumns[constants.TpIndex] == nil {
+		row.OutputColumns[constants.TpIndex] = "default"
 	}
```

Comment:
@cbruno10 still not keen on this idea of potential 3 different aspects of an index

Recommendation `vpc_id` or `default` if not set.

Reply:
I think being as specific as we can in `tp_index` is helpful, as it lets users know where the flow log was created, and then they can look for flow log entries related to that specific resource, e.g., `where tp_index = 'subnet-12345`. I think if we find it creates too many indexes or is too confusing, we can make a more static `tp_index` value.